### PR TITLE
[Java] File upload using new additional router API

### PR DESCRIPTION
**Pull request by renatocaval**

### CODE REVIEW

File: file-upload/file-upload-java-sbt/build.sbt [hunk 1]
```diff
@@ -15,11 +15,8 @@ lazy val `fileupload-api` = (project in file("fileupload-api"))
   )
 
 lazy val `fileupload-impl` = (project in file("fileupload-impl"))
-  .enablePlugins(LagomJava, PlayJava)
+  .enablePlugins(LagomJava)
   .disablePlugins(PlayLayoutPlugin)
-  .settings(
-    routesGenerator := InjectedRoutesGenerator
```
Comment (marcospereira) [inline file-upload/file-upload-java-sbt/build.sbt]: 👍

File: file-upload/file-upload-scala-sbt/fileupload-impl/src/main/scala/com/example/fileupload/impl/FileUploadRouter.scala [hunk 2]
```diff
@@ -0,0 +1,5 @@
+package com.example.fileupload.impl
+
+class FileUploadRouter {
```
Comment (marcospereira) [inline file-upload/file-upload-scala-sbt/fileupload-impl/src/main/scala/com/example/fileupload/impl/FileUploadRouter.scala]: Leftover?

File: file-upload/file-upload-scala-sbt/fileupload-impl/src/main/resources/routes [hunk 3]
```diff
@@ -1,3 +0,0 @@
-
```
Comment (marcospereira) [inline file-upload/file-upload-scala-sbt/fileupload-impl/src/main/resources/routes]: This is the Scala version. Not a problem but probably should be deleted in https://github.com/lagom/lagom-recipes/pull/67 instead.

File: file-upload/file-upload-scala-sbt/fileupload-impl/src/main/scala/com/example/play/controllers/FileUploadController.scala [hunk 4]
```diff
@@ -1,66 +0,0 @@
-package com.example.play.controllers
```
Comment (marcospereira) [inline file-upload/file-upload-scala-sbt/fileupload-impl/src/main/scala/com/example/play/controllers/FileUploadController.scala]: This is the Scala version. Not a problem but probably should be deleted in #67 instead.

**Comment by renatocaval** [comment]
@marcospereira, I started with one PR and then decided to split. When doing that I deleted too much. 

It's now fixed.

File: file-upload/file-upload-java-sbt/fileupload-api/src/main/java/com/example/fileupload/api/FileUploadService.java [hunk 5]
```diff
@@ -29,9 +30,7 @@ default Descriptor descriptor() {
                 .withCalls(
                         pathCall("/api/echo", this::uppercaseEcho)
                 )
-                .withServiceAcls(
-                        path("/api/echo"),
-                        path("/api/files")
-                );
+                .withAutoAcl(true)
+                .withServiceAcls(path("/api/files"));
```
Comment (ignasi35) [inline file-upload/file-upload-java-sbt/fileupload-api/src/main/java/com/example/fileupload/api/FileUploadService.java]: This ACL no longer exposes `"/api/echo"` via the gateway.

Comment (renatocaval) [reply file-upload/file-upload-java-sbt/fileupload-api/src/main/java/com/example/fileupload/api/FileUploadService.java]: have you tested it? I remember to have tested it and it was working. I may be wrong tough.

**Comment by renatocaval** [comment]
> Few comments I can't add as code reviews:
> 
>     1. Please bump `sbt.version` to `1.2.8` in `project/build.properties`.
> 
>     2. There are a few `warn` messages:

Can we do it on another PR as this is not related or introduced on this PR?

File: file-upload/file-upload-java-sbt/fileupload-api/src/main/java/com/example/fileupload/api/FileUploadService.java (hunk 5 above)
Comment (ignasi35) [reply file-upload/file-upload-java-sbt/fileupload-api/src/main/java/com/example/fileupload/api/FileUploadService.java]: You are right!

TIL https://github.com/lagom/lagom/blob/db77b62e66ba9ba725a25a3aaf09267b4e25037b/service/javadsl/api/src/main/java/com/lightbend/lagom/javadsl/api/Descriptor.java#L805

**Pull request merged**